src/shipit_code_coverage_backend/shipit_code_coverage_backend/services/active_data.py

```diff
@@ -0,0 +1,261 @@
+
+        query = ActiveDataCoverage.base_query(filters)
+        query.update({
+            # no search results please
+            'size': 0,
+
+            'aggs': {
+                'revisions': {
+                    'terms': {
+                        # List changeset ids, sorted by changeset_date
+                        'field': ActiveDataCoverage.FIELD_CHANGESET,
+                        'order': {'_date': 'desc'},
+                        'size': nb,
+                    },
+
+                    'aggs': {
+                        # Calc average changeset date per bucket (as timestamp)
+                        '_date': {
+                            'avg': {
+                                'field': ActiveDataCoverage.FIELD_CHANGESET_DATE,
+                                'missing': 0,
+                            },
+                        }
+                    },
+                },
+            },
+        })
+
+        async with ActiveDataClient() as es:
+            out = await es.search(
+                index=secrets.ACTIVE_DATA_INDEX,
+                body=query,
+            )
+            return out['aggregations']['revisions']['buckets']
+
+    @staticmethod
+    async def get_revision_date(changeset):
+        '''
+        Get the date of a revision
+        '''
+        query = ActiveDataCoverage.base_query([
+            {'term': {ActiveDataCoverage.FIELD_CHANGESET: changeset}},
+        ])
+        query.update({
+            # no search results please
+            'size': 0,
+
+            'aggs': {
+                'date': {
+                    'avg': {
+                        'field': ActiveDataCoverage.FIELD_CHANGESET_DATE,
+                    },
+                },
+            },
+        })
+        async with ActiveDataClient() as es:
+            out = await es.search(
+                index=secrets.ACTIVE_DATA_INDEX,
+                body=query,
+            )
+            return out['aggregations']['date']['value']
+
+    @staticmethod
+    async def calc_revision_coverage(changeset):
+        '''
+        Calculate total coverage in percent for a changeset
+        Directly done through an aggregation on ES
+        '''
+        query = ActiveDataCoverage.base_query([
+            {'term': {ActiveDataCoverage.FIELD_CHANGESET: changeset}},
+        ])
+        query.update({
+            # no search results please
+            'size': 0,
+
+            'aggs': {
+                'percentage': {
+                    'avg': {
+                        'field': ActiveDataCoverage.FIELD_TOTAL_PERCENT,
+                    },
+                },
+            },
+        })
+        async with ActiveDataClient() as es:
+            out = await es.search(
+                index=secrets.ACTIVE_DATA_INDEX,
+                body=query,
+            )
+            return out['aggregations']['percentage']['value']
+
+    @staticmethod
+    @alru_cache(maxsize=2048)
+    async def get_coverage(changeset):
```

Comment:
In the future, we might want to split this into two requests, one to get the current (very fast) and one to get the previous (slower).
But it depends on how the frontend is going to evolve.